addons/mail/models/mail_thread.py

```diff
@@ -3000,7 +3000,10 @@ def _message_compute_parent_id(self, parent_id):
                 [('res_id', '=', self.id), ('model', '=', self._name), ('message_type', '!=', 'user_notification')],
                 order='date desc, id desc',
                 limit=200,  # arbitrary, but sometimes loops / spam may creater a long history
-            ).sorted(lambda msg: (msg.message_type in ('comment', 'email'), msg.date, msg.id), reverse=True)
+            ).sorted(
+                lambda msg: (msg.message_type in ('comment', 'email'), msg.date or msg.create_date, msg.id),
```

Comment:
`create_date` is technically not required either (though it is probably always set in practice)

Reply:
Good time to fix those scripts then. create_date is often checked in code and we consider it to be set in a lot of code places, so we can start patching the whole codebase but I bet scripts should be fixed instead.